File: StartUpExample/StartUpExampleJava/app/src/main/java/com/tinlone/startupexamplejava/initializers/DatabaseProxyInitializer.java
package com.tinlone.startupexamplejava.initializers;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.startup.Initializer;

import com.tinlone.startupexamplejava.dependencies.database.DatabaseHelper;
import com.tinlone.startupexamplejava.dependencies.database.DatabaseProxy;

import java.util.Arrays;
import java.util.List;

public class DatabaseProxyInitializer implements Initializer<DatabaseProxy> {
    @NonNull
    @Override
    public DatabaseProxy create(@NonNull Context context) {
        Log.i("loglog", "create: DatabaseProxyInitializer");
        DatabaseProxy.initialize(DatabaseHelper.getInstance());
        return DatabaseProxy.getInstance();
    }

    @NonNull
    @Override
    public List<Class<? extends Initializer<?>>> dependencies() {
        return Arrays.asList(LoggerInitializer.class, DatabaseHelperInitializer.class);
    }
}

File: StartUpExample/StartUpExampleKotlin/app/src/main/java/com/tinlone/startupexamplekotlin/initializers/DatabaseProxyInitializer.kt
package com.tinlone.startupexamplekotlin.initializers

import android.content.Context
import android.util.Log
import androidx.startup.Initializer
import com.tinlone.startupexamplekotlin.dependencies.database.DatabaseHelper
import com.tinlone.startupexamplekotlin.dependencies.database.DatabaseProxy
import java.util.*

class DatabaseProxyInitializer : Initializer<DatabaseProxy?> {
    override fun create(context: Context): DatabaseProxy {
        Log.i("loglog", "create: DatabaseProxyInitializer")
        DatabaseProxy.initialize(DatabaseHelper.getInstance())
        return DatabaseProxy.getInstance()
    }

    override fun dependencies(): List<Class<out Initializer<*>?>> {
        return listOf(LoggerInitializer::class.java, DatabaseHelperInitializer::class.java)
    }
}